core/src/consensus/tendermint/mod.rs

```diff
@@ -480,36 +478,33 @@ impl Tendermint {
     }
 
     fn backup(&self) {
-        if let Some(c) = self.client.read().as_ref().and_then(|weak| weak.upgrade()) {
```

Comment:
It seems that it changes the behavior.
Is it okay?

Reply:
Yes. The client should be valid in all places.